src/lib/core/questions.ts

```diff
@@ -552,24 +553,55 @@ export const questionGroups: QuestionGroup[] = [
 			{
 				condition: [{ name: "features", contains: "adapter" }],
 				type: "select",
-				name: "adminReact",
-				label: "Admin with React",
-				message: "Use React for the Admin UI?",
-				initial: "no",
-				choices: ["yes", "no"],
+				name: "adminUi",
+				label: "Admin UI",
+				message:
+					"Which framework would you like to use for the Admin UI?",
+				initial: "json",
+				choices: [
+					{
+						message: "JSON UI",
+						hint: "(good for simple Admin UIs with a few fields)",
+						value: "json",
+					},
+					{
+						message: "HTML / Materialize",
+						hint: "(good for Admin UIs that have a few special requirements)",
+						value: "html",
+					},
+					{
+						message: "React",
+						hint: "(good for complex Admin UIs)",
+						value: "react",
+					},
+					{
+						message: "No UI",
+						hint: "(should only be used if you have another way to configure)",
+						value: "none",
+					},
+				],
+				replay: (answers: Record<string, any>): void => {
+					if (answers.adminReact === "yes") {
+						answers.adminUi = "react";
+					} else if (answers.adminReact === "no") {
+						answers.adminUi = "html";
+					}
+				},
 				migrate: async (ctx) =>
-					(await ctx.hasFilesWithExtension(
-						"admin/src",
-						".jsx",
-						(f) => !f.endsWith("tab.jsx"),
-					)) ||
-					(await ctx.hasFilesWithExtension(
-						"admin/src",
-						".tsx",
-						(f) => !f.endsWith("tab.tsx"),
-					))
-						? "yes"
-						: "no",
+					(await ctx.fileExists("admin/jsonConfig.json"))
+						? "json"
+						: (await ctx.hasFilesWithExtension(
+								"admin/src",
+								".jsx",
+								(f) => !f.endsWith("tab.jsx"),
+						  )) ||
+						  (await ctx.hasFilesWithExtension(
+								"admin/src",
+								".tsx",
+								(f) => !f.endsWith("tab.tsx"),
+						  ))
+						? "react"
+						: "html",
```

Comment:
I'd probably turn this into a proper function with if/else and return statements.